source/INI/PeanutButter.INI.Tests/TestINI.cs

```diff
@@ -1243,6 +1243,35 @@ public void ShouldRemoveTheSectionWhenFoundByName()
             }
         }
 
+        [TestFixture]
+        public class Renaming
+        {
+            [Test]
+            public void ShouldReturnOnNonExistentSection()
+            {
+                var sut = Create();
+                var section = GetRandomString(4);
+
+                // Assert
+                Expect(sut).Not.To.Have.Section(section);
+            }
+
+            [Test]
+            public void ShouldRenameSection()
+            {
+                var sut = Create();
+                var section = GetRandomString(4);
+                var newSection = GetRandomString(5);
+                sut.AddSection(section);
+                sut.SetValue(section, GetRandomString(2), GetRandomString(2));
+                sut.RenameSection(section, newSection);
+
+                // Assert
+                Expect(sut).Not.To.Have.Section(section);
+                Expect(sut).To.Have.Section(newSection);
```

Comment:
for completeness, this test should also assert that the new value is in the new section -- this would prove that the operation was a rename, and not an erroneous delete.

Reply:
ok